Add docs for the Debezium data connector & the CDC feature (#296)
spiceai/docs: spiceaidocs/docs/data-accelerators/data-refresh.md

      refresh_check_interval: 10m
```

## Changes

> Accelerations configured with `refresh_mode: changes` requires that the data connector supports Change Data Capture (CDC) and can return a stream of row-level changes. Currently the only supported data connector for this mode is [Debezium](../data-connectors/debezium.md).

Reviewer: ```suggestion
Accelerations configured with `refresh_mode: changes` requires that the data connector supports Change Data Capture (CDC) and can return a stream of row-level changes. The only supported data connector for this mode is [Debezium](../data-connectors/debezium.md).
```

Author: Closing in favor of Luke's suggestion

> Accelerations configured with `refresh_mode: changes` requires that the data connector supports Change Data Capture (CDC) and can return a stream of row-level changes. Currently the only supported data connector for this mode is [Debezium](../data-connectors/debezium.md).

Reviewer 2: ```suggestion
Datasets configured with acceleration `refresh_mode: changes` require a [Change Data Capture (CDC)](../features/cdc/index.md) supported data connector. Initial CDC support in Spice is supported by the [Debezium data connector](../data-connectors/debezium.md).
```

## Behavior on Zero Results

> By default, accelerated datasets will only return results that have been accelerated locally. If the locally accelerated data is a subset of the full dataset in the federated source, i.e. through setting `refresh_sql`, `refresh_data_window` or configuring retention policies, queries against the accelerated dataset may return zero results, where the federated table would return results. Supported for `full` and `append` refresh modes.
> 
> Control this behavior by setting `on_zero_results` in the acceleration configuration.

Reviewer 2: ```suggestion
By default, accelerated datasets only return locally materialized data. If this local data is a subset of the full dataset in the federated source—due to settings like `refresh_sql`, `refresh_data_window`, or retention policies—queries against the accelerated dataset may return zero results, even when the federated table would return results.

To address this, `on_zero_results: use_source` can be configured in the acceleration configuration. Queries returning zero results will fall back to the federated source, returning results from querying the underlying data.

The `on_zero_results: use_source` setting applies only to `full` and `append` refresh modes (not `changes).
```

## Refresh On-Demand

> Accelerated datasets can be refreshed on-demand via the `refresh` CLI command or `POST /v1/datasets/:name/acceleration/refresh` API endpoint. Supported for `full` and `append` refresh modes.

Reviewer 2: ```suggestion
Accelerated datasets can be refreshed on-demand via the `refresh` CLI command or `POST /v1/datasets/:name/acceleration/refresh` API endpoint.

On-demand refresh applies only to `full` and `append` refresh modes (not `changes).
```

## Refresh Retries

> By default, accelerated datasets attempt to retry data refreshes on transient errors (connectivity issues, compute warehouse goes idle, etc.) using [Fibonacci](https://en.wikipedia.org/wiki/Fibonacci_sequence) backoff strategy. This behavior can be adjusted with the [`acceleration.refresh_retry_enabled`](/reference/spicepod/datasets#accelerationrefresh_retry_enabled) and [`acceleration.refresh_retry_max_attempts`](/reference/spicepod/datasets#accelerationrefresh_retry_max_attempts) parameters. Supported for `full` and `append` refresh modes.

Reviewer 2: ```suggestion
By default, data refreshes for accelerated datasets are retried on transient errors (connectivity issues, compute warehouse goes idle, etc.) using [Fibonacci](https://en.wikipedia.org/wiki/Fibonacci_sequence) backoff strategy. 

Retry behavior can be configured using the [`acceleration.refresh_retry_enabled`](/reference/spicepod/datasets#accelerationrefresh_retry_enabled) and [`acceleration.refresh_retry_max_attempts`](/reference/spicepod/datasets#accelerationrefresh_retry_max_attempts) parameters.

Data refresh retry applies to `full` and `append` refresh modes not `changes` which inherently supports data integrity and consistency through the CDC mechanism.
```

## Retention Policy

> A retention policy automatically removes data from accelerated datasets with a temporal column that exceeds the defined retention period, optimizing resource utilization. Supported for `full` and `append` refresh modes.

Reviewer 2: ```suggestion
Accelerated datasets can be set to automatically evict time-series data exceeding a retention period by setting a retention policy based on the configured `time_column` and `acceleration.retention_period`.

Retention policies apply to `full` and `append` refresh modes (not `changes`).
```
